onyxia-api/src/main/java/fr/insee/onyxia/api/services/impl/HelmAppsService.java

```diff
@@ -324,6 +330,115 @@ public Service getUserService(Region region, Project project, User user, String
         return getHelmApp(region, user, result);
     }
 
+    @Override
+    public Service getUserServiceDetails(Region region, Project project, User user, String serviceId)
+            throws MultipleServiceFound, ParseException {
+        if (serviceId.startsWith("/")) {
+            serviceId = serviceId.substring(1);
+        }
+        String namespace = kubernetesService.determineNamespaceAndCreateIfNeeded(region, project, user);
+        HelmLs result =
+                getHelmInstallService()
+                        .getAppById(
+                                getHelmConfiguration(region, user),
+                                serviceId,
+                                namespace
+                                );
+        HelmReleaseInfo helmReleaseInfo =
+                getHelmInstallService()
+                        .getAll(
+                                getHelmConfiguration(region, user),
+                                result.getName(),
+                                result.getNamespace());
+        Service res = getHelmApp(region, user, result);
+        KubernetesClient client = kubernetesClientProvider.getUserClient(region, user);
+        List<HasMetadata> hasMetadata;
+        try (InputStream inputStream =
+                new ByteArrayInputStream(helmReleaseInfo.getManifest().getBytes(StandardCharsets.UTF_8))) {
+            hasMetadata = client.load(inputStream).items();
+        } catch (IOException e) {
+            throw new RuntimeException("Exception during loading manifest", e);
+        }
+        List<Object> manifests = new ArrayList<Object>(hasMetadata); 
+        res.setTemplates(manifests);
+        List<Map<String, Object>> podInfoList = new ArrayList<>();
+        
+        List<Pod> pods = client.pods().inNamespace(namespace).list().getItems();
+        for (Pod pod : pods) {
+            Map<String, Object> podInfo = new HashMap<>();
+            podInfo.put("podName", pod.getMetadata().getName());
+            podInfo.put("owners", getOwnerReferences(pod,client));
+            podInfoList.add(podInfo);
+        }
+        res.setPodsAndOwners(filterPodsByTopLevelOwnerReferences(podInfoList, hasMetadata, namespace, client));                                                             
+        return res;
+    }
+
+    private List<Map<String, Object>> getOwnerReferences(HasMetadata resource, KubernetesClient client) {
+        List<Map<String, Object>> owners = new ArrayList<>();
+
+        List<OwnerReference> ownerReferences = resource.getMetadata().getOwnerReferences();
+        if (ownerReferences != null) {
+            for (OwnerReference ownerReference : ownerReferences) {
+                String kind = ownerReference.getKind();
+                String name = ownerReference.getName();
+
+                // Fetch the owner resource
+                HasMetadata ownerResource = fetchOwnerResource(resource.getMetadata().getNamespace(), kind, name, client);
```

Comment:
Need to optimize this to prevent multiple fetching of the same resource.  
A cache local to the function should help a lot